microagents/README.md

```diff
@@ -0,0 +1,261 @@
+Each repository can have its own instructions in `.openhands/microagents/repo.md`. These instructions are:
+- Private to that repository
+- Automatically loaded when working with that repository
+- Perfect for repository-specific guidelines and team practices
+
+Example repository structure:
+```
+your-repository/
+└── .openhands/
+    └── microagents/
+        └── repo.md    # Repository-specific instructions
+        └── knowledges/  # Private micro-agents that are only available inside this repo
+        └── tasks/       # Private micro-agents that are only available inside this repo
+```
+
+
+## Loading Order
+
+When OpenHands works with a repository, it:
+1. Loads repository-specific instructions from `.openhands/microagents/repo.md` if present
+2. Loads relevant knowledge agents based on keywords in conversations
+3. Enable task agent if user select one of them
+
+## Types of MicroAgents
+
+All microagents use markdown files with YAML frontmatter.
+
+
+### 1. Knowledge Agents
+
+Knowledge agents provide specialized expertise that's triggered by keywords in conversations. They help with:
+- Language best practices
+- Framework guidelines
+- Common patterns
+- Tool usage
+
+Key characteristics:
+- **Trigger-based**: Activated by specific keywords in conversations
+- **Context-aware**: Provide relevant advice based on file types and content
+- **Reusable**: Knowledge can be applied across multiple projects
+- **Versioned**: Support multiple versions of tools/frameworks
+
+Example:
+
+```markdown
+---
+name: github
+version: 1.0.0
+agent: CodeActAgent
+triggers:
+- github
+- git
+---
+
+You have access to an environment variable, `GITHUB_TOKEN`, which allows you to interact with
+the GitHub API.
+
+You can use `curl` with the `GITHUB_TOKEN` to interact with GitHub's API.
+ALWAYS use the GitHub API for operations instead of a web browser.
+
+Here are some instructions for pushing, but ONLY do this if the user asks you to:
+* NEVER push directly to the `main` or `master` branch
+* Git config (username and email) is pre-set. Do not modify.
+* You may already be on a branch starting with `openhands-workspace`. Create a new branch with a better name before pushing.
+* Use the GitHub API to create a pull request, if you haven't already
+* Use the main branch as the base branch, unless the user requests otherwise
+* After opening or updating a pull request, send the user a short message with a link to the pull request.
+* Do all of the above in as few steps as possible. E.g. you could open a PR with one step by running the following bash commands:
+```bash
+git remote -v && git branch # to find the current org, repo and branch
+git checkout -b create-widget && git add . && git commit -m "Create widget" && git push -u origin create-widget
+curl -X POST "https://api.github.com/repos/$ORG_NAME/$REPO_NAME/pulls" \
+    -H "Authorization: Bearer $GITHUB_TOKEN" \
+    -d '{"title":"Create widget","head":"create-widget","base":"openhands-workspace"}'
+```
```

Comment:
To prevent the documentation from getting too overwhelming, and also to prevent things from getting out-of-date due to the same content being copied in multiple places, maybe we can just make this a link.

```suggestion
You can see an example of a knowledge-based agent in [OpenHands's github microagent](https://github.com/All-Hands-AI/OpenHands/tree/main/microagents/knowledge/github.md).
```